Shortcuts to adjust width size, number of results shown and game mode

onesounds opened the pull request:
<img src="https://user-images.githubusercontent.com/6903107/188794316-95c17592-c0a4-49e4-a43f-669a9700c43c.gif" width="400">

![image](https://user-images.githubusercontent.com/6903107/188794403-a433a783-bc0d-4025-81aa-642b5b8240f6.png)

- Quick Adjust Width Size by Ctrl + Plus/Minus in query box. 
- Quick Adjust Max Result Shown by Ctrl + [ , Ctrl + ] in query box. 
- Add Tooltip 
- Adjustment adhere to maximum and minimum allowed
- Ctrl + F12 to change to Game Mode

onesounds commented:
- Add Ctrl+F12 to Toggle Game Mode.

onesounds commented:
I changed it because it was more intuitive to change it like this.
- Plus Minus : Max Result
- square brackets : Width

taooceros commented on Flow.Launcher/MainWindow.xaml.cs:
```diff
@@ -504,16 +505,76 @@ private void OnKeyDown(object sender, KeyEventArgs e)
+                    if (specialKeyState.CtrlPressed)
+                    {
+                        
+                        _settings.WindowSize = _settings.WindowSize + 100;
```
I am curious why does this work. I suspect we need to set the property in the viewmodel to trigger change?

taooceros commented:
@onesounds please take a look on my latest change

onesounds commented:
> @onesounds please take a look on my latest change

It's well organized and the code works well.

onesounds commented:
but why the build faild?

Garulf commented:
Are we still using <kbd>Ctrl</kbd>+ Arrow keys?

taooceros commented:
> Are we still using <kbd>Ctrl</kbd>+ Arrow keys?
> If so this may interfere with normal text box shortcuts. 

I remove them.

jjw24 commented:
1. When you use the shortcut for expanding or shrinking the query window width, the new width doesn't seem to get reflected back in the settings window.
2. Expanding the width of the query window does not respect the maximum width in the Settings, consequently you can keep expanding the width forever
3. Using the shortcut to expand the result list doesn't respect the maximum allowed results displayed, consequently can keep expanding forever.

onesounds commented:
> 1. When you use the shortcut for expanding or shrinking the query window width, the new width doesn't seem to get reflected back in the settings window.
> 2. Expanding the width of the query window using the shortcut does not respect the maximum width in the Settings, consequently you can keep expanding the width forever
> 3. Using the shortcut to expand the result list doesn't respect the maximum allowed results displayed, consequently can keep expanding forever.

Fixed

onesounds commented:
Need to make sure the subtitle sentence is appropriate. @Garulf

jjw24 commented on Flow.Launcher/ViewModel/MainViewModel.cs:
```diff
@@ -21,10 +21,11 @@
 namespace Flow.Launcher.ViewModel
 {
-    public class MainViewModel : BaseModel, ISavable
+    public partial class MainViewModel : BaseModel, ISavable
```
@taooceros why is this changed to partial class?

taooceros replied:
Because I would like to use the mvvm toolkit to simplify the code of creating RelayCommand, which requires partial to generate additional code.

jjw24 commented on global.json:
```diff
@@ -1,6 +1,6 @@
-        "version": "6.0.100",
-        "rollForward": "latestFeature"
+        "version": "6.0.*",
+        "rollForward": "latestPatch"
```
@taooceros what's this change for?

taooceros replied:
Oh it shouldn't be there (I am testing different version of dotnet and see which one can help with the bug. However it is not needed because workaround is found)...Though it makes sense to apply the latest patch since that generally includes some security fixes.

jjw24 commented on Flow.Launcher/Flow.Launcher.csproj:
```diff
@@ -114,4 +115,14 @@
   <Target Name="PreBuild" BeforeTargets="PreBuildEvent">
     <Exec Command="taskkill /f /fi &quot;IMAGENAME eq Flow.Launcher.exe&quot;" />
   </Target>
+
+  <Target Name="RemoveDuplicateAnalyzers" BeforeTargets="CoreCompile">
+    <!-- Work around https://github.com/dotnet/wpf/issues/6792 -->
+
+    <ItemGroup>
+      <FilteredAnalyzer Include="@(Analyzer->Distinct())" />
+      <Analyzer Remove="@(Analyzer)" />
+      <Analyzer Include="@(FilteredAnalyzer)" />
+    </ItemGroup>
+  </Target>
```
What's this change?

jjw24 replied:
@taooceros What's this change for us specifically?

jjw24 commented:
> Need to make sure the subtitle sentence is appropriate. @Garulf

Updated the wording, hopefully ok.

taooceros commented:
> @taooceros What's this change for us specifically?

I couldn't reply directly. This is a workaround for source generator. Details here https://github.com/dotnet/wpf/issues/6792.

If that's being fixed, we can remove it.

taooceros commented:
> > @taooceros What's this change for us specifically?
> 
> I couldn't reply directly. This is a workaround for source generator. Details here [dotnet/wpf#6792](https://github.com/dotnet/wpf/issues/6792).
> 
> If that's being fixed, we can remove it.

I think it is fixed (in 6.0.401).

VictoriousRaptor commented:
Technically it's Ctrl + "=" and Ctrl + "-".

jjw24 commented:
👍 https://github.com/Flow-Launcher/Flow.Launcher/pull/1480

onesounds commented:
@jjw24  @VictoriousRaptor 

You guys are too programmers. Why don't you consider the average user? 
This is my intended to make it easier for the general user to memorize hotkey. 
I changed the shortcut once in the middle for this PR. 

+:increase count.
-: decrease count.
[ : decrease size
] : increase size

It is a key chosen so that it can be intuitively understood.

"Technically it's Ctrl + "=" and Ctrl + "-"." <-  This can be talked about. This is because it is not a strong suggestion.
"OK I changed it" <- NOoooooooo

![image](https://user-images.githubusercontent.com/6903107/196313645-9428a76e-f529-4667-a629-6566a64cba3b.png)

**Photoshop.** 
They are not idiots who don't even know the keycode.

![image](https://user-images.githubusercontent.com/6903107/196315653-6586eacd-74f7-4769-86e5-94f4a4e26450.png)
**memopad**

![image](https://user-images.githubusercontent.com/6903107/196316218-8dae1208-c43b-408c-867a-36207cbb841b.png)
**firefox**

I have never seen a case of writing = and - instead of + and - when matching these shortcuts. What stupid program writes shortcuts like this? flow launcher? 😂

jjw24 commented:
Lol looking at those screenshots, realised I never paid much attention to how those shortcuts are described, makes sense. Will keep it as it is.

VictoriousRaptor commented:
In fact I found it when was looking at my keyboard and realized that + needs a shift combined to input. And I opened powertoys,tried adding some hotykeys, found that they use "=" to describe that key. And I thought every software use "=". Really never paid attention to that.

onesounds commented:
good. You guys sins are forgiven.

<img src="https://user-images.githubusercontent.com/6903107/196320765-9cc9df03-5ab5-4022-88c2-038c8e7c41a6.png" width="250">

VictoriousRaptor commented:
So maybe for a single key, = is correct but for pairs like our new shortcut, + and - is a more user friendly choice.
